docs/ci-integration/ci-integration.md

```diff
@@ -0,0 +1,118 @@
+
+Earthly also has a special [`--push`](../earthfile/earthfile.md#--push) option that can be used when invoking a target. In a CI, you may want to ensure this flag is present to push images or run commands that are not typically done as part of a normal development workflow.
+
+If you would like to do cross-platform builds, you will need to install some [`binfmt_misc`](https://github.com/multiarch/qemu-user-static) entries. This can be done by running: `docker run --rm --privileged multiarch/qemu-user-static --reset -p yes`. This installs the needed entries and `qemue-user-static` binaries on your system. This will need to be repeated on each physical box (only once, since its a kernel level change, and the kernel is shared across containers).
+
+To share secrets with `earthly`, use the [`--secret`](../earthfile/earthfile.md#--secret-env-varsecret-ref) option to inject secrets into your builds. You could also use our [cloud secrets](../guides/cloud-secrets.md), for a more seamless experience.
+
+#### Networking & Security
+
+Upon invocation, `earthly` depends on the availability of an `earthly-buildkit` daemon to perform its build. This daemon has some networking and security considerations.
+
+Large builds can generate many `docker` pull requests for certain images. You can setup and use a [pull through cache](pull-through-cache.md) to circumvent this.
+
+If `earthly` is running on a dedicated host, the only consideration to take is the ability to run the container in a `--privileged` mode. Typical installations *should* support this out of the box. We also support running under user namespaces, [when `earthly` is configured to start the `earthly-buildkit` container with the `--userns host` option](../earthly-config/earthly-config.md#buildkit_additional_args). Rootless configurations are currently unsupported.
+
+If `earthly` is connecting to a remote `earthly-buildkitd`, then you will need to take additional steps. See this article for [running a remote buildkit instance](guides/remote-buildkit.md).
+
+### Examples
+
+Below are links to CI systems that we have produced more specific guides for. If you run into anything in your CI that wasn't covered here, we would love to add it to our documentation. Pull requests are welcome!
+
+ * [Jenkins](guides/jenkins.md)
+ * [Kubernetes](guides/kubernetes.md)
+ * [Circle CI](guides/circle-integration.md)
+ * [AWS CodeBuild](guides/codebuild-integration.md)
+ * [GitHub Actions](guides/gh-actions-integration.md)
```

Comment:
I feel that the structure of the previous version of this page was better because it had Step 1, step 2, step 3 etc. Wondering if this new version could be adapted to still have that structure. It makes it feel more like a finite process, rather than a possibly endless rabbit hole.

Reply:
As I have plugged it into more and more CIs, its not quite a step-by-step process. Each one requires some kind of bespoke work as I have branched out... the intent was to have guides per CI; rather than a set of global steps that likely miss something.

Reply:
I added more structure, and I think its a nice flow, particularly with the simplification. We lead by laying out phases, the guide follows it... but it doesn't fel forced or overly-simplistic, IMO.